src/main/java/com/squareup/javapoet/AnnotationSpec.java

```diff
@@ -183,13 +185,17 @@ public Builder toBuilder() {
   }
 
   @Override public boolean equals(Object o) {
-    return o instanceof AnnotationSpec
-        && ((AnnotationSpec) o).type.equals(type)
-        && ((AnnotationSpec) o).members.equals(members);
+    if (this == o)
+      return true;
+    if (o == null)
+      return false;
+    if (getClass() != o.getClass())
+      return false;
+    return hashCode == o.hashCode();
```

Comment:
Unequal strings can have equal hashcodes, since there is >32 bits in many strings. Hashcode is never suitable for `equals`.